Support GoBGP with API V3

Ales999 opened the pull request:
All test is ok.

```text
make -C ui/ test
make[1]: Entering directory '/home/ales999/go/src/github.com/ales999/alice-lg/ui'
yarn install
yarn install v1.22.22
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.47s.
yarn test --ci --all --watchAll=false
yarn run v1.22.22
$ react-scripts test --ci --all --watchAll=false
 PASS  src/app/components/routes/flags/RejectCandidateIndicator.test.js (9.815 s)
 PASS  src/app/components/routes/flags/PrimaryIndicator.test.js (9.851 s)
 PASS  src/app/components/routes/flags/RpkiIndicator.test.js (9.974 s)
 PASS  src/app/components/content/Content.test.js
 PASS  src/app/components/routes/FlagIcon.test.js
 PASS  src/app/components/datetime/RelativeTimestampFormat.test.js
 PASS  src/app/components/datetime/DateTime.test.js
 PASS  src/app/components/datetime/RelativeTimestamp.test.js
 PASS  src/app/components/datetime/time.test.js
 PASS  src/app/components/datetime/RelativeTime.test.js
 PASS  src/app/components/spinners/WaitingCard.test.js
 PASS  src/app/components/spinners/LoadingIndicator.test.js
 PASS  src/app/Main.test.js

Test Suites: 13 passed, 13 total
Tests:       16 passed, 16 total
Snapshots:   0 total
Time:        16.333 s
Ran all test suites.
Done in 17.62s.
make[1]: Leaving directory '/home/ales999/go/src/github.com/ales999/alice-lg/ui'
mkdir -p ./ui/build
touch ./ui/build/UI_BUILD_STUB
go test ./pkg/...
ok      github.com/alice-lg/alice-lg/pkg/api    0.006s
ok      github.com/alice-lg/alice-lg/pkg/caches 0.068s
ok      github.com/alice-lg/alice-lg/pkg/config 0.016s
ok      github.com/alice-lg/alice-lg/pkg/decoders       0.004s
ok      github.com/alice-lg/alice-lg/pkg/http   0.005s
ok      github.com/alice-lg/alice-lg/pkg/pools  0.002s
?       github.com/alice-lg/alice-lg/pkg/sources        [no test files]
ok      github.com/alice-lg/alice-lg/pkg/sources/birdwatcher    0.017s
?       github.com/alice-lg/alice-lg/pkg/sources/gobgp  [no test files]
ok      github.com/alice-lg/alice-lg/pkg/sources/gobgp/apiutil  0.009s
ok      github.com/alice-lg/alice-lg/pkg/sources/openbgpd       0.005s
ok      github.com/alice-lg/alice-lg/pkg/store  0.018s
ok      github.com/alice-lg/alice-lg/pkg/store/backends/memory  0.275s
ok      github.com/alice-lg/alice-lg/pkg/store/backends/postgres        0.875s
rm ./ui/build/UI_BUILD_STUB
make: *** No rule to make target '...'.  Stop.
```

Update api code to using last version GoBGP (v3.34.0)

annikahannig commented:
Hej! Thank you so much for your work! I am currently review this and will merge this later! :)

Ales999 commented:
Sorry, I need some more time to finalize it, after more testing.

Ales999 commented:
Despite everything working fine and all tests passing, I am confused by this:
![image](https://github.com/user-attachments/assets/0768cb5f-dadb-40b6-ab62-0a3442fe9c25)
I don't understand JS very well and couldn't find what the problem is. It's probably not related to the new API.
I also successfully tested store_backend as postgres, although it is much slower on my VM than using only memory to store routes.
```bash
$ make test
make -C ui/ test
make[1]: Entering directory '/home/ales999/go/src/github.com/ales999/alice-lg/ui'
yarn install
yarn install v1.22.22
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.40s.
yarn test --ci --all --watchAll=false
yarn run v1.22.22
$ react-scripts test --ci --all --watchAll=false
 PASS  src/app/components/routes/flags/PrimaryIndicator.test.js
 PASS  src/app/components/routes/flags/RejectCandidateIndicator.test.js
 PASS  src/app/components/routes/flags/RpkiIndicator.test.js
 PASS  src/app/components/routes/FlagIcon.test.js
 PASS  src/app/components/datetime/DateTime.test.js
 PASS  src/app/components/datetime/RelativeTime.test.js
 PASS  src/app/Main.test.js
 PASS  src/app/components/spinners/LoadingIndicator.test.js
 PASS  src/app/components/datetime/RelativeTimestampFormat.test.js
 PASS  src/app/components/content/Content.test.js
 PASS  src/app/components/datetime/time.test.js
 PASS  src/app/components/datetime/RelativeTimestamp.test.js
 PASS  src/app/components/spinners/WaitingCard.test.js

Test Suites: 13 passed, 13 total
Tests:       16 passed, 16 total
Snapshots:   0 total
Time:        2.693 s
Ran all test suites.
Done in 3.27s.
make[1]: Leaving directory '/home/ales999/go/src/github.com/ales999/alice-lg/ui'
mkdir -p ./ui/build
touch ./ui/build/UI_BUILD_STUB
go test ./pkg/...
ok      github.com/alice-lg/alice-lg/pkg/api    0.018s
ok      github.com/alice-lg/alice-lg/pkg/caches 0.078s
ok      github.com/alice-lg/alice-lg/pkg/config 0.013s
ok      github.com/alice-lg/alice-lg/pkg/decoders       0.010s
ok      github.com/alice-lg/alice-lg/pkg/http   0.006s
ok      github.com/alice-lg/alice-lg/pkg/pools  0.009s
?       github.com/alice-lg/alice-lg/pkg/sources        [no test files]
ok      github.com/alice-lg/alice-lg/pkg/sources/birdwatcher    0.024s
?       github.com/alice-lg/alice-lg/pkg/sources/gobgp  [no test files]
ok      github.com/alice-lg/alice-lg/pkg/sources/gobgp/apiutil  0.009s
ok      github.com/alice-lg/alice-lg/pkg/sources/openbgpd       0.004s
ok      github.com/alice-lg/alice-lg/pkg/store  0.024s
ok      github.com/alice-lg/alice-lg/pkg/store/backends/memory  0.335s
ok      github.com/alice-lg/alice-lg/pkg/store/backends/postgres        0.614s
rm ./ui/build/UI_BUILD_STUB
```

annikahannig commented:
I did quite some optimization on the memory store backend; postgres is just likely to be slower and will likely use more diskspace. The backend seemed like a good idea then; however in memory has become much more efficent and less resourcehungry - so I would recommend using this.

In regards to the status error:

https://github.com/alice-lg/alice-lg/blob/fb90181192d1fd1d7bdfe011dd1d9baed8f671bb/pkg/sources/gobgp/source.go#L139

I think there might be something missing.

annikahannig commented:
I just checked:

```go
type Status struct {
	ServerTime   time.Time `json:"server_time"`
	LastReboot   time.Time `json:"last_reboot"`
	LastReconfig time.Time `json:"last_reconfig"`
	Message      string    `json:"message"`
	RouterID     string    `json:"router_id"`
	Version      string    `json:"version"`
	Backend      string    `json:"backend"`
}
```

Only `RouterID` and `Backend`
```go
	response.Status.RouterID = resp.Global.RouterId
	response.Status.Backend = "gobgp"
```
are set.

So in order to fix the UI, I think the other fields need to also contain sensible values.
(Message might be something like `Daemon up and running` or something...)

Ales999 commented:
Yes, I will check and try to fix UI. Thanks for the tip.

annikahannig commented:
I'm currently reviewing you changes! :)

Ales999 commented:
My changes are few - in the end I just replaced the 'apiutil' directory from the gobgp sources (gobgp/pkg/apiutil).
It was not possible to quickly fix UI, since the API does not provide such data on versions and dates, and gobgp itself needs to be modified. The only thing I managed to get was this message, but I replaced it with a simple designation of the autonomous system. The date is not quite correct, but at least it is not noticeable.

annikahannig commented:
Yeah. I see. I should maybe add a more clear indicator in the UI if a datapoint is actually not available.
I think I saw a hardcoded interface name in there. (Not introduced by you.)

annikahannig commented:
Finally managed to review your MR. This looks good so far. I am resolving a couple of merge conflicts.

annikahannig commented:
Hmm looks like I can not mark this as merged *sigh*

Ales999 commented:
> Hmm looks like I can not mark this as merged _sigh_
Hm, I had to make changes from development. I think it would be easier to delete this one and create a new one where I immediately deleted the "iputil" folder based on how it is done in the client:
https://github.com/osrg/gobgp/blob/f733438a965b33813b23200ea10adfa1939f5a36/cmd/gobgp/monitor.go#L28
This is a completely working option, I have already tried it in my test client:
https://github.com/Ales999/TestGoBGP/blob/main/bgpclient.go#L14
